### ISSUE TITLE
Cross compilation with target x86_64-pc-windows-gnu fails to link

### ISSUE THREAD

**Issue by bantonsson** [libp2p/rust-libp2p#1447]
When cross compiling with target `x86_64-pc-windows-gnu` on a Linux box our resulting binary fails to link with this issue:

```
error: linking with `x86_64-w64-mingw32-gcc` failed: exit code: 1
  |
  = note: "x86_64-w64-mingw32-gcc" "-fno-use-linker-plugin" ...<lots of arguments removed>
  = note: /usr/bin/x86_64-w64-mingw32-ld: cannot find -lIphlpapi
          collect2: error: ld returned 1 exit status
```

This is because the `get_if_addrs` crate is defining its dependency on the `Iphlpapi` library like this `#[link(name = "Iphlpapi")]` instead of `#[link(name = "iphlpapi")]`.

Anyway the [`get_if_addrs`](https://github.com/maidsafe/get_if_addrs) repository is archived, so I tried to switch out `get_if_addrs` for [`pnet_datalink`](https://github.com/libpnet/libpnet.git) in the code but that pulled in `winpcap` which I don't think is necessary.

For now I've created a patched version of `get_if_addrs` here Actyx/get_if_addrs@2344800 that we use in our build to override it.

**Comment by romanb** [comment]
This may be fixed by https://github.com/libp2p/rust-libp2p/pull/1788 (see also https://github.com/messense/if-addrs/pull/4).

**Comment by romanb** [comment]
Should be resolved by #1788 and the `0.29` release. Feel free to reopen otherwise.
